jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/SerializedInvoker.java

```diff
@@ -188,10 +197,13 @@ public void run()
                     LOG.debug("Running link {} of {}", link, SerializedInvoker.this);
                 try
                 {
+                    recursionCounter++;
```

Comment:
Can this be just a boolean?

Reply:
Even better: this should be a `Thread` reference as we want to check if the current thread is the one invoking. Let me change that.